lib/origen/utility/collector.rb

```diff
@@ -1,14 +1,78 @@
 module Origen
   module Utility
     class Collector
-      attr_reader :store
+      attr_reader :_hash_
+      attr_reader :merge_method
 
-      def initialize
-        @store = {}
+      # Need to keep a seperate methods list so we know what's been added by method missing instead of what's
+      # been added either by the hash or by method missing.
+      # Only overwriting a value in the block should cause an error. Overriding a value from the hash depends on
+      # the merge method's setting.
+      attr_reader :_methods_
+
+      def initialize(options = {}, &block)
+        @merge_method = options[:merge_method] || :keep_hash
+        @fail_on_empty_args = options[:fail_on_empty_args]
+        unless [:keep_hash, :keep_block, :fail].include?(@merge_method)
+          fail Origen::OrigenError, "Origen::Utility::Collector cannot merge with method :#{@merge_method} (of class #{@merge_method.class}). Known merge methods are :keep_hash (default), :keep_block, or :fail"
+        end
+
+        @_hash_ = options.key?(:hash) ? options[:hash].clone : {}
+        @_methods_ = []
+
+        if block_given?
+          yield self
+        end
+      end
+
+      # Legacy store method.
+      def store
+        Origen.log.deprecate 'Collector::store method was used. Please use the Ruby-centric Collector::to_h or Collector::to_hash method instead' \
+                             " Called from: #{caller[0]}"
```

Comment:
I think if you call `Origen.deprecate` instead of the direct log method, it should report the caller for you like this, just FYI

Reply:
Thanks! I'll use this next time. I didn't know the Origen module had that method as well.